drivers/can/can_stm32.c

```diff
@@ -409,6 +410,46 @@ static int can_stm32_init(const struct device *dev)
 		return -EIO;
 	}
 
+	/* configure pinmux */
+	if (cfg->pinctrl_len != 0U) {
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32f1_pinctrl)
+		int remap;
+		/* Check that remap configuration is coherent across pins */
+		remap = stm32_dt_pinctrl_remap_check(cfg->pinctrl,
+						     cfg->pinctrl_len);
+		if (remap < 0) {
+			return remap;
+		}
+
+		/* A valid remapping configuration is provided */
+		/* Apply remapping before proceeding with pin configuration */
+		LL_APB2_GRP1_EnableClock(LL_APB2_GRP1_PERIPH_AFIO);
+
+		switch ((uint32_t)cfg->can) {
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(can1), okay)
+		case DT_REG_ADDR(DT_NODELABEL(can1)):
+			if (remap == REMAP_FULL) { /* or REMAP_1 ??? */
+				LL_GPIO_AF_EnableRemap_CAN1();
+			} else {
+				LL_GPIO_AF_DisableRemap_CAN1();
+			}
+			break;
+#endif
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(can2), okay)
+		case DT_REG_ADDR(DT_NODELABEL(can2)):
+			if (remap == REMAP_FULL) { /* or REMAP_1 ??? */
```

Comment:
`EnableRemap_XXX` functions refer to the `REMAP_FULL` case. Note that availability of these LL API calls are conditional to `AFIO_MAPR_CAN2_REMAP` being defined.

Reply:
@erwango how should we handle cases where LL API calls are not available for all F1 families? Check for the same define as LL API? Or using Zephyr definitions like `CONFIG_SOC_STM32F107XC`...?

Reply:
One idea is to use the HAL defines directly
#ifdef AFIO_MAPR_CAN_REMAP_REMAP1 and AFIO_MAPR_CAN2_REMAP.

Reply:
Thanks for your comments, @gmarull and @erwango . Implemented it directly using the HAL defines now.